### PR TITLE
TUM-30-files-list-frontend

### PR DESCRIPTION
**Pull request by ketrab100**

### CODE REVIEW

File: Frontend/Tumultu/src/app/file/api/index.ts
Comment (PaterMajkel) [inline Frontend/Tumultu/src/app/file/api/index.ts:1]: you don't have to export files like that in an angular project. Simple export interface in a model file is more than enough

File: Frontend/Tumultu/src/app/file/api/models/response/index.ts
Comment (PaterMajkel) [inline Frontend/Tumultu/src/app/file/api/models/response/index.ts:1]: same as earlier

File: Frontend/Tumultu/src/app/file/list/components/files-list/files-list.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<mat-list style="width: 90%">
+  <div *ngFor="let file of files | async">
```
Comment (PaterMajkel) [inline Frontend/Tumultu/src/app/file/list/components/files-list/files-list.component.html]: just an Idea, we can implement new angular syntax for iterations in .html file:

@for (file of files | async; ) {
  <mat-list-item>....
}

New syntax provides easier methods to implement empty list alerts and many different useful templates

File: Frontend/Tumultu/src/app/file/list/components/files-list/files-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  standalone: true,
+  imports: [NgForOf, AsyncPipe, FileItemComponent, MatList, MatListItem],
+  providers: [FilesFacade, FilesStore, FilesService],
+  templateUrl: './files-list.component.html',
+  styleUrl: './files-list.component.scss',
+})
+export class FilesListComponent {
+  files = this.fileFacade.getFiles();
+  constructor(private fileFacade: FilesFacade) {
+    console.log();
```
Comment (PaterMajkel) [inline Frontend/Tumultu/src/app/file/list/components/files-list/files-list.component.ts]: leftover console.log

File: Frontend/Tumultu/src/app/file/shared/stores/files.store.ts
Comment (PaterMajkel) [inline Frontend/Tumultu/src/app/file/shared/stores/files.store.ts:1]: I think it's better to export an observable in getFiles and when updating the store we would just use files$$.next(files)
right now after seting behaviourSubject we would be never able to gather an updated value

**Review by PaterMajkel — CHANGES_REQUESTED** [review]
Change some things so it would be easier to work on it in the future

File: Frontend/Tumultu/src/app/file/shared/stores/files.store.ts
Comment (ketrab100) [reply Frontend/Tumultu/src/app/file/shared/stores/files.store.ts:1]: subject should be modified only inside the store https://stackoverflow.com/questions/73648114/why-is-it-recommended-to-make-subjects-in-your-service-private-and-just-expose

File: Frontend/Tumultu/src/app/file/api/index.ts
Comment (ketrab100) [reply Frontend/Tumultu/src/app/file/api/index.ts:1]: Barrel files just simplify imports

**Review by PaterMajkel — APPROVED** [review]

**Pull request merged**